main.tf

```diff
@@ -342,3 +342,52 @@ resource "null_resource" "confirm_network_healthy" {
     }
   }
 }
+
+
+resource "ibm_container_addons" "addons" {
+  count             = var.ignore_worker_pool_size_changes && !(var.disable_public_endpoint) ? 1 : 0
+  cluster           = ibm_container_vpc_cluster.autoscaling_cluster[0].id
+  resource_group_id = var.resource_group_id
+
+  dynamic "addons" {
+    for_each = var.addons
+    content {
+      name    = addons.key
+      version = addons.value
+    }
+  }
+}
+
+resource "time_sleep" "wait_operators" {
+  depends_on      = [ibm_container_addons.addons]
+  create_duration = "5s"
+}
+
+locals {
+  worker_pool_config = [
+    for worker in var.worker_pools :
+    {
+      name    = worker.pool_name
+      minSize = worker.minSize
+      maxSize = worker.maxSize
+      enabled = worker.enableAutoscaling
+    } if worker.enableAutoscaling != null && worker.minSize != null && worker.maxSize != null
+  ]
+
+}
+
+resource "kubernetes_config_map_v1_data" "set_autoscaling" {
+  count      = var.ignore_worker_pool_size_changes && !(var.disable_public_endpoint) ? 1 : 0
+  depends_on = [ibm_container_addons.addons, time_sleep.wait_operators]
```

Comment:
Best practise is to minimise the number of explicit depends on.

In this case it is sufficient to depend on time_sleep, since time_sleep also explicitly depends on ibm_container_addons.